app/oracle-server/src/main/scala/org/bitcoins/oracle/server/OracleRoutes.scala

```diff
@@ -195,7 +195,13 @@ case class OracleRoutes(oracle: DLCOracle)(implicit
           complete {
             oracle.signEvent(oracleEventTLV, EnumAttestation(outcome)).map {
               eventDb =>
-                Server.httpSuccess(eventDb.sigOpt.get.hex)
+                val oracleEvent = OracleEvent.fromEventDbs(Vector(eventDb))
+                oracleEvent match {
+                  case _: PendingOracleEvent =>
+                    throw new RuntimeException("Failed to sign event")
```

Comment:
Why are we throwing here? Shouldn't we be using `Server.httpError()` or `reject()`? 

This applies in other places too in this file.

Reply:
throwing will cause a `Server.httpError`

Reply:
Also, looks like we can't replace it with a `Server.httpError`, it causes an compiler error

Reply:
I think throwing here is fine, but if you were interested in the marginally more efficient solution it would be using `flatMap` in place of `map` and then using `Future.failed` and `Future.successful` as this throw is being caught by the future and passed up, but catching takes more overhead than flattening